aiohttp/client.py

```diff
@@ -608,9 +608,7 @@ async def _request(
                             parsed_url = url.join(parsed_url)
 
                         is_same_host_https_redirect = (
-                            url.host == parsed_url.host
-                            and parsed_url.scheme == "https"
-                            and url.scheme == "http"
+                            url.host == parsed_url.host and parsed_url.scheme == "https"
```

Comment:
My suggestion was on the if statement below. This variable can be completely removed if we go that way.